src/data/markdown/docs/30 xk6-browser/01 Get started/02 Running xk6-browser.md

```diff
@@ -0,0 +1,245 @@
+    page.locator('input[name="password"]').type('123');
+
+    page.close();
+    browser.close();
+  }
+  ```
+
+  </CodeGroup>
+
+The preceding code creates and returns a Locator object with the selectors for both login and password passed as arguments. 
+
+Within the Locator API, various methods such as `type()` can be used to interact with the elements. The `type()` method types a text to an input field. 
+
+## Asynchronous operations
+
+For some context, k6 operations are synchronous. However, Playwright operations are asynchronous in nature and since xk6-browser aims to provide rough compatibility with Playwright, some of our operations are now asynchronous.
+
+At the moment, `page.waitForNavigation()` and `Element.click()` return promises and work similarly to [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises).
+
+To avoid timing errors or other race conditions in your script, if you have actions that load up a different page, you need to make sure that you wait for that action to finish before continuing.
+
+  <CodeGroup labels={["script.js"]} lineNumbers={[true]}>
+
+  ```javascript
+  import { check } from 'k6';
+  import { chromium } from 'k6/x/browser';
+
+  export default function () {
+    const browser = chromium.launch({ headless: false });
+    const page = browser.newPage();
+
+    page.goto('https://test.k6.io/my_messages.php', { waitUntil: 'networkidle' });
+
+    // Enter login credentials and login
+    page.locator('input[name="login"]').type('admin');
+    page.locator('input[name="password"]').type('123');
+
+    // Wait for asynchronous operations to complete
+    // page.$() is used instead of page.locator
+    // because page.locator does not support asynchronous click yet
+    // eslint-disable-next-line no-undef
+    Promise.all([page.waitForNavigation(), page.$('input[type="submit"]').click()])
+      .then(() => {
+        check(page, {
+          header: page.locator('h2').textContent() == 'Welcome, admin!',
+        });
+      })
+      .finally(() => {
+        page.close();
+        browser.close();
+      });
+  }
+  ```
+
+  </CodeGroup>
+
+The preceding code uses `Promise.all([])` to wait for the two promises to be resolved before continuing. Since clicking the submit button causes page navigation, `page.waitForNavigation()` is needed because the page won't be ready until the navigation completes. This is required because there can be a race condition if these two actions don't happen simultaneously. 
+
+Then, you can use [`check`](https://k6.io/docs/javascript-api/k6/check/) from the k6 API to assert the text content of a specific element. Finally, you close the page and the browser.
+
+## Assertions
+
+You can add assertions in your browser script via:
+1. Checks, as mentioned in the previous section
+2. [k6chaijs](https://k6.io/docs/javascript-api/jslib/k6chaijs/) for a more BDD (Behavior Driven Development) or TDD (Test Driven Development) style
```

Comment:
@mdcruz I think we should not recommend `k6chaijs` at this moment and stick with "official" APIs. 

We have very little documentation about k6chaijs. The 99% of our assertion examples use Checks. 

Introducing `k6chaijs` in this article adds additional overhead.

Reply:
Makes sense Pepe, I'll remove this section then 👍